$MirOS: ports/www/firesomething/patches/patch-security_coreconf_OpenBSD_mk,v 1.4 2008/12/08 22:38:53 tg Exp $
$OpenBSD: patch-security_coreconf_OpenBSD_mk,v 1.3 2005/04/09 15:43:47 wilfried Exp $
--- security/coreconf/OpenBSD.mk.orig	Thu Nov 22 04:39:33 2007
+++ security/coreconf/OpenBSD.mk	Tue Mar 16 18:27:24 2010
@@ -20,6 +20,7 @@
 # the Initial Developer. All Rights Reserved.
 #
 # Contributor(s):
+# * The MirOS Project
 #
 # Alternatively, the contents of this file may be used under the terms of
 # either the GNU General Public License Version 2 or later (the "GPL"), or
@@ -56,7 +57,7 @@ OS_LIBS			+= -pthread
 DSO_LDOPTS		+= -pthread
 endif
 
-DLL_SUFFIX		= so.1.0
+DLL_SUFFIX		= so.${SO_VERSION}
 
 OS_CFLAGS		= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -ansi -Wall -Wno-switch -pipe -DOPENBSD
 
@@ -65,8 +66,8 @@ OS_LIBS			= 
 ARCH			= openbsd
 
 DSO_CFLAGS		= -fPIC -DPIC
-DSO_LDOPTS		= -shared -fPIC -Wl,-soname,lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-DSO_LDFLAGS		=
+DSO_LDOPTS		+= -shared -fPIC -Wl,-soname,lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX) -Bsymbolic
+DSO_LDFLAGS		+= $(LDFLAGS)
 
 MKSHLIB			= $(CC) $(DSO_LDOPTS)
 
